src/expr/src/scalar/func.rs

```diff
@@ -2550,10 +2554,10 @@ impl BinaryFunc {
                 ScalarType::Int64 => contains_range_elem::<i64>(a, b),
                 ScalarType::Date => contains_range_elem::<Date>(a, b),
                 ScalarType::Numeric { .. } => contains_range_elem::<OrderedDecimal<Numeric>>(a, b),
-                ScalarType::Timestamp => {
+                ScalarType::Timestamp { .. } => {
                     contains_range_elem::<CheckedTimestamp<NaiveDateTime>>(a, b)
                 }
-                ScalarType::TimestampTz => {
+                ScalarType::TimestampTz { precision: None } => {
```

Comment:
Why the inconsistency here?

Reply:
Unless I am understanding this wrong, it should be same as above. Will update it to be so.
cc @benesch